hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/dao/LabelRequestConverter.java

```diff
@@ -8,15 +9,27 @@
 
 public class LabelRequestConverter {
   CreateLabelRequest convertCreationRequest(LabelCreateRequest creationRequest) {
-    return CreateLabelRequest.newBuilder()
-        .setData(LabelData.newBuilder().setKey(creationRequest.label().key()).build())
-        .build();
+    LabelData.Builder dataBuilder = LabelData.newBuilder().setKey(creationRequest.label().key());
+    if (Objects.nonNull(creationRequest.label().color())) {
+      dataBuilder.setColor(Objects.requireNonNull(creationRequest.label().color()));
+    }
+    if (Objects.nonNull(creationRequest.label().description())) {
+      dataBuilder.setDescription(Objects.requireNonNull(creationRequest.label().description()));
+    }
+    return CreateLabelRequest.newBuilder().setData(dataBuilder.build()).build();
   }
 
   UpdateLabelRequest convertUpdateRequest(LabelUpdateRequest updateRequest) {
+    LabelData.Builder dataBuilder = LabelData.newBuilder().setKey(updateRequest.label().key());
+    if (Objects.nonNull(updateRequest.label().color())) {
```

Comment:
common conversion code

Reply:
The labels are of different types. In create request, it is of type `Create Label` and in update request, it is of type `Label`. How can we combine them?

Reply:
I thought `label()` was of the same type and we could have a `Label` -> `LabelData` converter, but looking below in the review it looks like CreateLabel and Label duplicate the fields - which is OK, we had discussed why we can't break that.

What we _can_ do however, is extract an interface out of the two and put all the common fields in them (similar to how `Identifiable` works). This is actually a better solution than the `LabelData` pattern we use in the proto message - there we're stuck with composition since proto doesn't support inheritance. It doesn't introduce awkward nesting, it's backwards compatible, and allows for shared converters like I'm suggesting here.

So I'd suggest extracting a common interface from Label and CreateLabel (e.g. `LabelData`), put the common fields on that, extend it in the existing Label and CreateLabel interfaces, and then use it to convert here.

Reply:
@aaron-steinfeld I don't think this is a blocker. Can we address it in separate PR?

Reply:
yes, agreed it's not a blocker